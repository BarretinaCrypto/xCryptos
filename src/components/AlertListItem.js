import { func } from 'prop-types';
import { Text, View } from 'react-native';
import React from 'react';
import { C, STYLE } from '../config';
import Touchable from './Touchable';
import styles from './AlertListItem.style';

const AlertListItem = (props) => {
  const { alert, onPress } = props;
  const { low, high } = alert;

  return (
    <Touchable onPress={onPress}>
      <View style={[STYLE.ROW, STYLE.LIST_ITEM]}>
        <View style={styles.left}>
          <View style={STYLE.ROW}>
            <Text style={styles.symbol}>$</Text>
            <Text style={styles.price}>{low.toFixed(2)}</Text>
          </View>
          <Text style={styles.label}>low</Text>
        </View>
        <View style={styles.right}>
          <View style={STYLE.ROW}>
            <Text style={styles.symbol}>$</Text>
            <Text style={styles.price}>{high.toFixed(2)}</Text>
          </View>
          <Text style={styles.label}>high</Text>
        </View>
      </View>
    </Touchable>
  );
};

AlertListItem.propTypes = {
  alert: C.SHAPE.ALERT,
  onPress: func,
};

AlertListItem.defaultProps = {
  alert: {},
  onPress: undefined,
};

export default AlertListItem;
