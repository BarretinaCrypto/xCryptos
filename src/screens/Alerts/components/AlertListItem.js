import { func, shape } from 'prop-types';
import { Text, View } from 'react-native';
import React from 'react';
import { Amount, Touchable } from '../../../components';
import { SHAPE, STYLE } from '../../../config';
import styles from './AlertListItem.style';

const { ALERT } = SHAPE;

const AlertListItem = (props) => {
  const { alert, onPress } = props;
  const { low, high } = alert;

  return (
    <Touchable onPress={onPress}>
      <View style={[STYLE.ROW, STYLE.LIST_ITEM]}>
        <View style={styles.left}>
          <Amount style={styles.price} value={low} />
          <Text style={styles.label}>low</Text>
        </View>
        <View style={styles.right}>
          <Amount style={styles.price} value={high} />
          <Text style={styles.label}>high</Text>
        </View>
      </View>
    </Touchable>
  );
};

AlertListItem.propTypes = {
  alert: shape(ALERT),
  onPress: func,
};

AlertListItem.defaultProps = {
  alert: {},
  onPress: undefined,
};

export default AlertListItem;
