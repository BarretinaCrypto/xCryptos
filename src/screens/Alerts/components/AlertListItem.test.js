import 'react-native';
import React from 'react';
import AlertListItem from './AlertListItem';
import renderer from 'react-native-test-utils';

jest.mock('TouchableOpacity', () => 'TouchableOpacity');
jest.mock('../../../components/Amount', () => 'Amount');

const ALERT = { high: 10, low: 5 };

describe('<AlertListItem />', () => {
  it('renders by default', () => {
    const tree = renderer(<AlertListItem />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders high and low props', () => {
    const tree = renderer(<AlertListItem alert={ALERT} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('passes the onPress prop on the Touchable component', () => {
    const onPress = () => {};
    const component = renderer(<AlertListItem onPress={onPress} />);

    expect(component.query('TouchableOpacity').props.onPress).toEqual(onPress);
  });
});
