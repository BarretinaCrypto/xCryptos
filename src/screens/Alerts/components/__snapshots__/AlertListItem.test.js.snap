// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlertListItem /> renders by default 1`] = `
<TouchableOpacity
  onPress={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
        },
        Object {
          "backgroundColor": "#ffffff",
          "borderBottomColor": "rgba(0, 0, 0, 0.1)",
          "borderBottomWidth": 1,
          "paddingHorizontal": 16,
          "paddingVertical": 10,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Amount
        style={
          Object {
            "fontSize": 20,
          }
        }
        value={undefined}
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.5)",
            "fontSize": 12,
          }
        }
      >
        below
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
        }
      }
    >
      <Amount
        style={
          Object {
            "fontSize": 20,
          }
        }
        value={undefined}
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.5)",
            "fontSize": 12,
          }
        }
      >
        above
      </Text>
    </View>
  </View>
</TouchableOpacity>
`;

exports[`<AlertListItem /> renders high and low props 1`] = `
<TouchableOpacity
  onPress={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
        },
        Object {
          "backgroundColor": "#ffffff",
          "borderBottomColor": "rgba(0, 0, 0, 0.1)",
          "borderBottomWidth": 1,
          "paddingHorizontal": 16,
          "paddingVertical": 10,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Amount
        style={
          Object {
            "fontSize": 20,
          }
        }
        value={5}
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.5)",
            "fontSize": 12,
          }
        }
      >
        below
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
        }
      }
    >
      <Amount
        style={
          Object {
            "fontSize": 20,
          }
        }
        value={10}
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.5)",
            "fontSize": 12,
          }
        }
      >
        above
      </Text>
    </View>
  </View>
</TouchableOpacity>
`;
